### ISSUE TITLE
User needs to be authenticated to call this API

### ISSUE THREAD

**Issue by JoseeWouters** [aws-amplify/amplify-ui#4898]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Next.js

### Please describe your bug.

I have a NextJS 14 app and I want to create a mobile app with Capacitor, so I need a static/client side app. In my browser I can login just fine, but when I open it in the Xcode simulator i get the following error: "User needs to be authenticated to call this API". 
I don't understand the problem. 

I use the `<Authenticator />` in a component like advised here: https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#using-third-party-packages-and-providers and I then import this in app/layout.tsx. 

I created a `<ConfigureAmplifyClientSide />` component like advised here: https://docs.amplify.aws/nextjs/build-a-backend/server-side-rendering/nextjs/#configure-amplify-library-for-client-side-usage (and then the part meant for the App Router application).

### What's the expected behaviour?

To be able to login

### Help us reproduce the bug!

When opening this in Xcode Simulator or on an iPhone, the problem occurs

### Code Snippet

```typescript
// Put your code below this line.

```

### Console log output

There are no console messages

**Comment by ioanabrooks** [comment]
Hi @JoseeWouters, please provide more details. Have you configured the backend, set up authentication and called `Amplify.configure`  as indicated [here](https://docs.amplify.aws/nextjs/build-a-backend/auth/set-up-auth/) and [here](https://ui.docs.amplify.aws/react-native/connected-components/authenticator)?

**Comment by JoseeWouters** [comment]
Yes, the backend works, I've done all those steps. in the browser everything works perfectly. The problem occurs in the Xcode simulator and on the iPhone. 
For this I export the static build as instructed by Next.js and run this according to the instructions of Capacitor.

**Comment by rastoelias** [comment]
I'm encountering a similar issue with the **withAuthenticator** HOC in a Next.js 14 app integrated with AWS Amplify gen 2. However, my experience is slightly different:

- The error occurs exclusively in Safari. Chrome, Firefox, and Opera do not exhibit this issue.
- The error message appears when attempting to sign in or create an account.
- Interestingly, when creating an account, after submitting the confirmation code, I receive the error message, but the user is still created and confirmed successfully in AWS Cognito.

This leads me to believe that the issue might be related to how Safari handles sessions, cookies, or specific security policies that differ from other browsers. Here's a brief overview of my setup and the error:

### Setup:

- Next.js 14 with App Router.
- AWS Amplify gen 2.
- Using withAuthenticator for authentication.

**Comment by JoseeWouters** [comment]
That's an interesting theory, Xcode simulator is also Safari. 

I remade my app in Vue, because that worked before without any problems. It was still on V5, I migrated to V6 and tried this in Xcode and Safari, works out of the box. 

So I now think it's indeed related to Safari, but in combination with Next.js.

**Comment by storcswap** [comment]
I had the same problems. Came to this thread and can confirm it is specific to Safari. I tried debugging myself. 

Can anyone confirm it's just for sandbox? I don't want to get further to find out it doesn't work for prod either.

**Comment by calebpollman** [comment]
@rastoelias @storcswap Based on your comments here it sounds like a different issue than the issue @JoseeWouters is encountering as the projects are using Amplify Gen2 (rather than migrating from Amplify JS v5 to v6). Suggest opening a new issue in the [Amplify Backend repo](https://github.com/aws-amplify/amplify-backend)

**Comment by calebpollman** [comment]
@JoseeWouters Apologies for the churn here. Transferring this issue over to the Amplify JS team for further triage since it reads as an issue with usage of NextJS App Router and Capacitor, not the `Authenticator` itself
